py/server/deephaven_internal/stream.py

```diff
@@ -0,0 +1,87 @@
+import io
+
+
+class TeeStream(io.TextIOBase):
+    """TextIOBase subclass that splits output between a delegate instance and a set of lambdas.
+
+    Can delegate some calls to the provided stream but actually write only to the lambdas. Useful to adapt any existing
+    file-like object (such as sys.stdout or sys.stderr), but actually let some other processing take place before
+    writing.
+    """
+
+    @classmethod
+    def split(cls, py_stream, java_stream):
+        return TeeStream(
+            orig_stream=py_stream,
+            should_write_to_orig_stream=True,
+            write_func=lambda t: java_stream.write(bytes(t, py_stream.encoding)),
+            flush_func=lambda: java_stream.flush(),
+            close_func=lambda: java_stream.close()
+        )
+
+    @classmethod
+    def redirect(cls, py_stream, java_stream):
+        if hasattr(py_stream, "encoding"):
+            encoding = py_stream.encoding
+        else:
+            encoding = 'UTF-8'
```

Comment:
I'm curious why we need to check `hasattr` here, but not for `split`?

Reply:
Should be for both, thanks, was missing, since jpy doesn't break "split" case (nor have other writers i've encountered broken it).